src/uct/ib/ud/ud_mlx5.h

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2014.  ALL RIGHTS RESERVED.
+ *
+ * $COPYRIGHT$
+ * $HEADER$
+ */
+#ifndef UD_MLX5_H
+#define UD_MLX5_H
+
+#include <uct/ib/base/ib_verbs.h>
+#include <uct/ib/mlx5/ib_mlx5.h>
+#include <uct/ib/mlx5/ib_mlx5_log.h>
+
+#include "ud_iface.h"
+#include "ud_ep.h"
+#include "ud_def.h"
+
+typedef struct {
+    uct_ud_ep_t          super;
+    struct mlx5_wqe_av   av;
+} uct_ud_mlx5_ep_t;
+
+typedef struct {
+    uct_ud_iface_t        super;
+    struct {
+        uct_ib_mlx5_txwq_t  wq; 
+        uct_ib_mlx5_cq_t    cq;
+    } tx;
+    struct {
+        uct_ib_mlx5_rxwq_t  wq; 
+        uct_ib_mlx5_cq_t    cq;
+    } rx;
+} uct_ud_mlx5_iface_t;
+
+
+static inline unsigned uct_ud_mlx5_tx_moderation(uct_ud_mlx5_iface_t *iface)
```

Comment:
for ud, we can do a better check instead of checking a counter. can check if sw_pi crossed a boundary.